pyhomematic/devicetypes/actors.py

```diff
@@ -372,6 +372,37 @@ def ELEMENT(self):
         return [4]
 
 
+class IPKeySwitchLevel(GenericDimmer, GenericSwitch, HMEvent, HelperActionPress, HelperActorLevel):
+    """
+    Switch with two independent controllable LEDs, turning plugged in device on or off.
+    """
+    def __init__(self, device_description, proxy, resolveparamsets=False):
+        super().__init__(device_description, proxy, resolveparamsets)
+
+        self.WRITENODE.update({"STATE": [4],
+                               "LEVEL": [8, 12]})
+        self.EVENTNODE.update({"PRESS_SHORT": [1, 2],
+                               "PRESS_LONG": [1, 2]})
+
+    def on(self, channel=None):
+        """Turn light/switch on."""
+        if channel in self.WRITENODE["LEVEL"]:
+            self.set_level(1.0, channel)
+        else:
+            self.set_state(True, channel)
+
+    def off(self, channel=None):
+        """Turn light/switch off."""
+        if channel in self.WRITENODE["LEVEL"]:
+            self.set_level(0.0, channel)
+        else:
+            self.set_state(False, channel)
+
+    @property
+    def ELEMENT(self):
+        return self.WRITENODE["STATE"] + self.WRITENODE["LEVEL"]
```

Comment:
```suggestion
        return [4, 8, 12]
```

Reply:
Hard-coding the values instead of using the dictionary got rid of my error.
I don't have time now to keep debugging; I'll try to find more time later on, but no promises ;-)